### DISCUSSION TITLE
Question about carrier loss error / standalone config (Sideband Mac)

### DISCUSSION THREAD

**Discussion by cobraPA** [markqvist/Reticulum#277] [Help & Questions] [answered]
I'm experimenting with Sideband / Reticulum in Mac and running Sideband from git in a local python environment.  These are my first times trying to start a local instance of a Reticulum client and on launch I see the application and it seems to run properly.  In console, I am seeing messages like:

```
[Notice] Launching platform-specific service for RNS and LXMF
[Notice] Purging stale messages... /Users/cobra//.config/sideband/app_storage/sideband.db
[Warning] AutoInterface[Default Interface] Detected possible carrier loss on en6: [Errno 65] No route to host
[Warning] AutoInterface[Default Interface] Detected possible carrier loss on en6: [Errno 65] No route to host

```

This is with the default config file.  The machine has both ethernet and Wifi enabled.  There are no other Reticulum clients active on launch.  Is this expected if no peers are available?  I reviewed raw manual Interfaces and Networks sections, so it seems like AutoInterface should be ok, but the 'No route to host' portion made me wonder if I needed something else configured or if there was a problem with Mac or my install of Sideband.

One more question, for testing purposes, is it possible to run two clients on the same machine?  I'm thinking of the Mac client and one of the handheld clients, like emulated Android or Simulated iOS?

Thanks in advance,
Kevin

**Comment by faragher** [comment]
I don't have a Mac and use NomadNet on non-mobile devices, but naively I'd check your config file to make sure your nodes are defined correctly *and* that you can see them in other applications. 

> There are no other Reticulum clients active on launch. Is this expected if no peers are available?

In my experience, yes. Unless you have an address and request it, you're waiting on announcements. Those roll out slowly. If you're not seeing another node in 30 - 60 minutes, something's wrong. I would suggest trying NomadNet first, as that allows a static page to served, meaning all you need is an address and a trusted peer to get confirmation the local system is working.

>One more question, for testing purposes, is it possible to run two clients on the same machine? I'm thinking of the Mac client and one of the handheld clients, like emulated Android or Simulated iOS?

I tripped over this myself. Each program or instance has its own stack (I don't know it well enough to use the right terminology) but they cooperate. 

Example: A is started. B is started. B looks for an instance and sees A. They cooperate seamlessly. C is started and also sees A and everything is great. B halts and nobody notices. However, if A halts, C no longer has communication. It's suggested if you run multiple Reticulum apps, you start a daemon first, which then serves each other program that can drop in and out without damaging the ecosystem.

As far as iOS, my understanding is it's impossible from the ground up. iOS is incompatible with services of this type running in the background and all notifications go through Apple servers, either linking the app to Apple or reducing functionality.

I am not part of the dev team, I sell the information for what I paid for it, but I hope this helps in the meantime.

**Reply by cobraPA** [reply]
> I don't have a Mac and use NomadNet on non-mobile devices, but naively I'd check your config file to make sure your nodes are defined correctly _and_ that you can see them in other applications.
> 

Yeah, that was really my question, I guess.  The default config has one AutoInterface.  From the docs it sounds like that should allow it to connect to others on the local net?  I don't yet have anything else running, but was going to try one of the mobile clients possibly (but simulated on the same machine.)

I'm unclear on how 'the other nodes' get defined, in this case.

**Reply by faragher** [reply]
That's a soft area for me. I've never run more than one Linux Reticulum machine on a subnet at once. Since (as far as I know) AutoInterface doesn't work on Windows, I have no experience with it. 

The situation I described happens automatically on a local machine, however. 

I don't have the defaults in front of me, but you define network endpoints in your reticulum config file like so:

```
# TCP/IP interface to the Dublin hub
[[RNS Testnet Dublin]]
type = TCPClientInterface
enabled = yes
target_host = dublin.connect.reticulum.network
target_port = 4965
# TCP/IP interface to the Frankfurt hub
[[RNS Testnet Frankfurt]]
type = TCPClientInterface
enabled = yes
target_host = frankfurt.connect.reticulum.network
target_port = 5377

```
 
By adding these, you'll end up connecting to the test net. If you're already connected, the system will record any new announces on the net. You should be able to see them in the announce stream. 

I've sent a few announces to the test net, but you should get some randomly when you're online. If you let me know (in GMT) when you intend to be on and testing, I can send some manually, but I think you're best off just walking away for a few hours with it running. That was my first experience. (Maybe we should set up something that announces much more frequently for the test net, I don't know)

**Reply by cobraPA** [reply]
Thank you, that was helpful...  I think maybe Dublin was down yesterday or I didn't wait long enough.  I tried it then and didn't see anything.  With both testnets added, I had a flurry of announce activity this morning, so basic functionality is working!  Also, with both testnets configured today, I did not see any "carrier loss" messages at all.  So I'll have to try with one or the other to see if that is a factor.

**Reply by cobraPA** [reply]
I saw an announce from you and sent a quick message as well.  Looks like client it working good with the testnet nodes defined.

**Reply by faragher** [reply]
Glad to hear it. The long delay to see a node or peer was something that confused me at first, too. It helped me to set up an in-house propagation node that Sideband uses by default. I really should make sure my test net isn't linking to the RNS test net via autointerface. Sometimes it just spams announces and/or messages across my RNode. No need to spam the test net.

**Comment by cobraPA** [comment]
So, the good news is yes, you can run a handheld device simulator and communicate with the desktop MacOS app.  Got it partly working today and sent a message between the two instances.

**Comment by cobraPA** [comment, marked as answer]
And now I tested with MacOS client and AutoInterface only.  It is able to communicate with the simulated client locally.  It does show the errors above until the second local client is running.  Then they seem to stop.  And as a teaser...  I'm testing with an iOS client.

**Reply by faragher** [reply]
Neat! If you can find a way to make it work on iOS, that'd really help adoption.

**Reply by markqvist** [reply]
Cool! What you're doing here is very interesting :) Looking forward to seeing where it goes!
